src/solana/base_client.rs

```diff
@@ -0,0 +1,209 @@
+// Solana base client trait with default implementations for Solana-based clients
+use async_trait::async_trait;
+use solana_sdk::{
+    instruction::Instruction,
+    pubkey::Pubkey,
+    signature::Signature,
+    signer::Signer,
+    system_instruction,
+    transaction::Transaction,
+};
+use std::str::FromStr;
+use log::{info, warn};
+
+use super::signing_client::SolanaSigningClient;
+use crate::common::transaction::TransactionResponse;
+
+/// Base client trait with default implementations for Solana-based clients.
+/// 
+/// For chains which are somehow unique in their common module implementations,
+/// these function definitions can be overridden to match the custom chain logic.
+#[async_trait]
+pub trait SolanaBaseClient: SolanaSigningClient {
+    /// Convert SOL to lamports
+    fn sol_to_lamports(sol: f64) -> u64 {
+        (sol * 1_000_000_000.0) as u64
+    }
+    
+    /// Convert lamports to SOL
+    fn lamports_to_sol(lamports: u64) -> f64 {
+        lamports as f64 / 1_000_000_000.0
+    }
+    
+    /// Transfer SOL with amount in SOL (not lamports)
+    async fn transfer_sol_amount(
+        &self,
+        to: &str,
+        amount_sol: f64,
+    ) -> anyhow::Result<TransactionResponse> {
+        let amount_lamports = Self::sol_to_lamports(amount_sol);
+        self.transfer_sol(to, amount_lamports).await
+    }
```

Comment:
I think this could sit under `SolanaSigningClient`, next to the `transfer_sol` impl